Find a functionally equivalent code:android.os.Bundle.hasFileDescriptors:COMMENT
Method Modifier: public      
Comment:/**
 * Reports whether the bundle contains any parcelled file descriptors.
 */

@@ -26,32 +26,32 @@
                 } else if (obj instanceof SparseArray) {
                     SparseArray<? extends Parcelable> array = (SparseArray<? extends Parcelable>) obj;
                     for (int n = array.size() - 1; n >= 0; n--) {
-                        if ((array.get(n).describeContents() & Parcelable.CONTENTS_FILE_DESCRIPTOR) != 0) {
+                        if ((array.valueAt(n).describeContents() & Parcelable.CONTENTS_FILE_DESCRIPTOR) != 0) {
                             fdFound = true;
                             break;
                         }
@@ -34,40 +34,40 @@
                 } else if (obj instanceof ArrayList) {
                     ArrayList array = (ArrayList) obj;
                     // Parcelables; only look inside for Parcelables
-                    if ((array.size() > 0) && (array.get(0) instanceof Parcelable)) {
+                    if (!array.isEmpty() && (array.get(0) instanceof Parcelable)) {
                         for (int n = array.size() - 1; n >= 0; n--) {
                             Parcelable p = (Parcelable) array.get(n);
                             if (p != null && ((p.describeContents() & Parcelable.CONTENTS_FILE_DESCRIPTOR) != 0)) {

